File: TP01/TP01/aquecimento.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>


#define NUMENTRADAS 1000
#define LINHA 1000

bool isUpper(char letra)
{
    return('A' <= letra && letra <= 'Z');
}//end isUpper
bool isFim (char *FIM)
{
    bool resultado = false;
    int tamanhoString = strlen(FIM);
    if(FIM != NULL && tamanhoString >= 3)
    {
       if(FIM[0] == 'F' && FIM[1] == 'I' && FIM[2] == 'M')
       {
           resultado = true;
       }//end if
    }//end if
    
    return (resultado);
}//end isFim
int quantasMaiusculas(char *palavra)
{
    int contador = 0;
    int tamanhoPalavra = 0;
    if(palavra != NULL && palavra != "")
    {
        tamanhoPalavra = strlen(palavra);
        for(int i = 0; i < tamanhoPalavra; i++)
        {
            if(isUpper(palavra[i]))
            {
                contador++;
            }//end if
        }//end for
    }//end if

    return (contador);
}//end quantasMaiusculas

int main (int argc, char** argv)
{
    //declarar variaveis
    char palavra[NUMENTRADAS][LINHA];
    int numEntradas = 0; //contador para as entradas
    int numMaiusculas = 0;

    //receber a primeira string como input
    fgets(palavra[numEntradas], LINHA, stdin);

    //receber as demais strings como input
    while(!(isFim(palavra[numEntradas])))
    {
        numEntradas++;
        fgets(palavra[numEntradas], LINHA, stdin);
    }

    //contar as maiusculas presentes em cada linha de entrada
    for(int i = 0; i < numEntradas; i++)
    {
        numMaiusculas = quantasMaiusculas(palavra[i]);
        printf("%d\n", numMaiusculas);
    }

    //liberar memoria alocada dinamicamente
    free(palavra);

    //fim do programa
    return 0;
}

File: TP01/TP01/aquecimento.java
import java.util.*;
public class aquecimento
{
    public static boolean isUpper(final char letra) {
        return ('A' <= letra && letra <= 'Z');
        // retorna true se o caracter for uma letra maiusculas e false se nao for
    }

    public static boolean FIMentrada(final String FIM) {
        boolean resultado = false;
        if (FIM.length() >= 3) {
            if (FIM.charAt(0) == 'F' && FIM.charAt(1) == 'I' && FIM.charAt(2) == 'M') {
                resultado = true;
            }
        }
        return (resultado);

    }

    public static int QuantasMaiusculas(final String palavra) {
        // declaracao de variavel
        int nMaiusculas = 0;

        if (palavra != "" && !(FIMentrada(palavra))) // se a palavra nao estiver vazia, prosseguir
        {
            final int tamanhoPalavra = palavra.length();
            /*
             * atribui o tamanho da palavra a uma variavel para evitar a repeticao da mesma
             * funcao varias vezes no for
             */

            for (int i = 0; i < tamanhoPalavra; i++) {
                if (isUpper(palavra.charAt(i))) // chama a funcao que testa se o char e letra maiuscula
                {
                    nMaiusculas++;
                }
            }
        }
        return (nMaiusculas);
    }

    public static void main(final String[] args) {
        // declarar variaveis
        int numeroEntradas = 0;
        int resposta = 0;
        final String[] entradas = new String[1000];

        // LEMBRAR DE MODIFICAR!!!!!!!!!!!!!!
        final Scanner leitor = new Scanner(System.in);
        
        //ler primeira entrada
        entradas[numeroEntradas] = leitor.nextLine();
        
        //contar quantas linhas de entrada e armazena-las no array de strings
        while(!(FIMentrada(entradas[numeroEntradas])))
        {
            numeroEntradas++;
            entradas[numeroEntradas] = leitor.nextLine();
        }

        //contar as letras maiusculas de cada linha de entrada valida
        for(int i = 0; i < numeroEntradas; i++)
        {
            resposta = QuantasMaiusculas(entradas[i]);
            System.out.println(resposta);
        }
    }
}